hack/cpaas/patch-cluster-with-rhpm-0.1.sh

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+set -eu -o pipefail
+
+PROJECT_ROOT="$(git rev-parse --show-toplevel)"
+declare -r PROJECT_ROOT
+declare -r powermon_package_name="power-monitoring-operator-bundle-container"
+declare -r OCP_VERSION=${OCP_VERSION:-'v4.13'}
+
+source "$PROJECT_ROOT/hack/utils.bash"
+
+function check_prereq {
+  header "checking pre requisites..."
+  if ! command -v jq &> /dev/null
+  then
+      err "jq could not be found. Please install jq first."
+      exit
+  fi
+  if ! command -v oc &> /dev/null
+  then
+      err "oc could not be found. Please install oc first."
+      exit
+  fi
+  if ! command -v podman &> /dev/null
+  then
+      err "podman could not be found. Please install podman first."
+      exit
+  fi
+  ok "all prereqs ok.."
+}
+
+function get_index_image {
+  local index_image="noimage"
+  local package=$1
+  local requested_ocp_version=$2
+  messages=$(curl -s 'https://datagrepper.engineering.redhat.com/raw?topic=/topic/VirtualTopic.eng.ci.redhat-container-image.index.built&delta=824000&contains='$package)
+
+  local i=0
+  local bundle=""
+  while true
+  do
+    message=$(echo $messages | jq --argjson i $i '.raw_messages[$i]')
+    bundle=$(echo $message | jq -r '.msg.index.added_bundle_images[0]')
+    version=$(echo $bundle| awk -F':' '{print $2}')
+	ocp_version=$(echo $message | jq -r '.msg.index.ocp_version')
+    if [[ $version == 0.1* ]] && [[ $ocp_version == "$requested_ocp_version" ]];
+    then
+      index_image=$(echo $message | jq -r '.msg.index.index_image')
+      break
+    else
+      i=$(expr $i + 1)
+    fi
+  done
+  ok "index image found: $index_image"
+  brew_index_image=$(echo $index_image | awk -F':' '{print "brew.registry.redhat.io/rh-osbs/iib:"$2}')
+  echo $brew_index_image
+}
+
+function add_brew_registry_credentials(){
+  header "Getting credentials for brew registry"
+  info "getting token from employee-token-manager"
+  local tokens=$(curl --negotiate -u : https://employee-token-manager.registry.redhat.com/v1/tokens -s)
+  if [[ "$?" != "0" ]] || [[ "$tokens" == "null" ]] || [[ "$tokens" == "" ]];
+  then
+	err "could not get token. \n\
+    please use the following command to create a token and retry the script. \n\
+    curl --negotiate -u : -X POST -H 'Content-Type: application/json' --data '{\"description\":\"for testing cpaas built powermon images on openshift 4 cluster\"}' https://employee-token-manager.registry.redhat.com/v1/tokens -s
+	"
+	exit -1
```

Comment:
maybe return 1 and capture the status code in the main function

Reply:
there is no point in continuing without this step.